### PR TITLE
mlton: fix build

### PR DESCRIPTION
**Pull request by gebner**
###### Motivation for this change

Failed to build because `which` is missing.
https://hydra.nixos.org/build/126588345/nixlog/1

ZHF: #97479

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/mlton/default.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, fetchgit, patchelf, gmp }:
+{ stdenv, fetchurl, fetchgit, patchelf, gmp, which }:
 rec {
   mlton20130715 = import ./20130715.nix {
```
Comment (worldofpeace) [inline pkgs/development/compilers/mlton/default.nix:3]: shouldn't these actually be `callPackage`?

Comment (gebner) [reply pkgs/development/compilers/mlton/default.nix]: Good point.  I've changed it.

**Comment by worldofpeace** [comment]
Thanks @gebner :sparkles: You know what do next :grimacing:

**Comment by gebner** [comment]
Backported as 3a8c2b75c2fa14cd46dc68dbdf89b34862bcf94b and a824b76b791fb178e1ee7a0769f1f805a7e48378.

**Pull request merged**